frontend/src/pages/LineageView.tsx

```diff
@@ -115,29 +115,90 @@ class LineageView extends React.Component<LineageViewProps, LineageViewState> {
     );
   }
 
-  public async refresh(): Promise<void> {
-    // Todo: Implement this!
+  private buildArtifactCards(artifacts: Artifact[]): CardDetails[] {
+    return [
+      {
+        title: 'Artifact',
+        elements: artifacts.map((artifact) => ({
+            resource: artifact,
+            prev: true,
+            next: true
+          })
+        )
+      }
+    ];
   }
 
-  private async reload(request: ListRequest): Promise<string> {
-    // TODO: Consider making an Api method for returning and caching types
-    // if (!this.artifactTypes || !this.artifactTypes.size) {
-    //   this.artifactTypes = await this.getArtifactTypes();
-    // }
-    // const {artifacts} = this.state;
-    // if (!artifacts.length) {
-    //   try {
-    //     const response = await this.api.metadataService.listArtifacts2();
-    //     this.setState({artifacts: (response && response.artifacts) || []});
-    //     this.clearBanner();
-    //   } catch (err) {
-    //     this.showPageError('Unable to retrieve Artifacts.', err);
-    //   }
-    // }
-    // this.setState({
-    //   rows: this.getRowsFromArtifacts(request),
-    // });
-    return '';
+  private buildExecutionCards(executions: Execution[]): CardDetails[] {
+    return [
+      {
+        title: 'Execution',
+        elements: executions.map((execution) => ({
+            resource: execution,
+            prev: true,
+            next: true
+          })
+        )
+      }
+    ];
+  }
+
+  private async loadData(id: number): Promise<string> {
+    // TODO: Optimize async calls with Promise.all()
+    const getEventsByArtifactIDsRequest = new GetEventsByArtifactIDsRequest();
+    getEventsByArtifactIDsRequest.addArtifactIds(id);
+
+    const getEventsByArtifactIDsResponse =
+      await this.metadataStoreService.getEventsByArtifactIDs(getEventsByArtifactIDsRequest);
+
+    const events = getEventsByArtifactIDsResponse.getEventsList();
+
+    const outputExecutionIds =
+      events.filter(isOutputEvent).map((event) => (event.getExecutionId()));
+    const inputExecutionIds =
+      events.filter(isInputEvent).map((event) => (event.getExecutionId()));
+
+    const outputExecutions = await this.getExecutions(outputExecutionIds);
```

Comment:
Let's parallelize this:
```suggestion
    const [outputExecutions, inputExecutions] = await Promise.all([
        this.getExecutions(outputExecutionIds),
        this.getExecutions(inputExecutionIds),
    ])
```

Reply:
Done.